### ISSUE TITLE
Fileclass Table View not working

### ISSUE THREAD

**Issue by kearalong** [mdelobelle/metadatamenu#181]
The new file class table view only seems to only work for me on file classes and notes that are mapped to a single tag.

Fileclasses that are classified in the front matter (I have it set to 'type:') do not display in the file class table view, and file classes where the tags are defined in the front matter ('tagNames:') don't display in the table view. 
Fileclasses mapped to a nested tag (The file class is 'source' but the files are 'source/[read status]') don't show in table view either. 
I've tested repeatedly with all of these styles and combos to try to figure this out. The only ones that work are a single tag mapped- if I add a second type to a file that is showing up in the table view, it will disappear from the table view. The file classes themselves are working, because the icons and properties match their file classes- but the table views always say no results for query.

**Comment by mdelobelle** [comment]
Hi! thank you for the report. It has been fixed in 0.3.7. Is it working for you now?

**Comment by Krakor92** [comment]
I have the exact same problem, all the fileclass table views inside my vault doesn't work at all (even after upgrading to 0.3.7).

So i decided to create a brand new vault to try to locate the issue and i think i found it:

Without changing any settings, it works as shown in your videos. But then i changed the `**fileClass field alias**` parameter inside the plugin settings and the table views stopped working.

Also, class instances do not populate the table view when defined with a `#class`

**Comment by mdelobelle** [comment]
I can't reproduce the issue by changing the fileclass field alias setting.
Maybe have you experienced an edge case when changing this setting (fieldIndex not properly refreshed?...), but after a plugin disable/enable it should work as expected

Did you see any error message in the console?

**Comment by Krakor92** [comment]
Unfortunately, no error can be found in the console 😕.
I tried disabling/enabling the plugin + clearing the dataview cache + reloading my vault without saving but it didn't change anything. I thought it might be my linux laptop's fault but I enabled syncing and it happens on my other devices too (an android phone and an iPad).

Anyway, it's not that big a problem, I just need to stick to `fileClass` and not rely on tag mapping features to make it work.

**Comment by mdelobelle** [comment]
Can you paste some samples of your vault?
- the settings part for fileClass
- a fileClass note
- a note using this fileClass

**Comment by Krakor92** [comment]
Sure ! Here is my minimal reproducible example and the steps i took to get it:
(It was done on a Windows 10 laptop and on Obsidian v1.0.3)

1. I created a new vault and named it '🚮' (since it's a throwaway vault) and opened the devtools
2. Turned off restricted mode
3. Installed Dataview first (v.0.5.47) and enabled it (with the DataviewJS queries too)
4. Installed Metadata Menu (v.0.3.8) and enabled it
5. Next, i created a `_classes` folder at the root of my vault.
	1. Inside i put an empty `Media` file
6. Set Metadata Menu settings class folder to `_classes`
7. Then i reloaded the app without saving because it couldn't detect that my `Media` was a class file
8. After that i've created a file at the root of my vault and named it `Arcane`
	1. After creating this file, returning to `Media` and jumping back to the new file, the plugin detected that `Media` was a class file
	2. So I right clicked on the `Arcane` file and selected the `Add fileClass to Arcane` command to choose `Media`

9. I then pressed on the new class icon next to `Media` in the outliner and it showed me the Media table view with my file in it (screen below)

![image](https://user-images.githubusercontent.com/24924824/203035537-9386d530-ad74-4b05-8a14-359890d257ae.png)


10. Without doing anything else, i've gone back to the metadata menu settings to change the `fileClass field alias` to `fileclass` instead of `fileClass`
11. Then i changed the yaml frontmatter property inside `Arcane` to `fileclass` and without having to reload the Table view, the query returned nothing (screen below)

![image](https://user-images.githubusercontent.com/24924824/203035600-6add5744-0977-40ad-a75a-9503a30d6d6b.png)

As for the file you asked for, here is the data.json from the metadata plugin and the data.json from the dataview plugin (I don't think it's useful to share the other two since you see them on the screens)

// metadata-menu/data.json
```json
{
  "presetFields": [],
  "fileClassQueries": [],
  "displayFieldsInContextMenu": true,
  "globallyIgnoredFields": [],
  "classFilesPath": "_classes/",
  "isAutosuggestEnabled": true,
  "fileClassAlias": "fileclass",
  "settingsVersion": 3,
  "firstDayOfWeek": 1,
  "enableLinks": true,
  "enableTabHeader": true,
  "enableEditor": true,
  "enableBacklinks": true,
  "enableStarred": true,
  "enableFileExplorer": true,
  "enableSearch": true,
  "buttonIcon": "clipboard-list",
  "tableViewMaxRecords": 100
}
```

---

// dataview/data.json
```json
{
  "renderNullAs": "\\-",
  "taskCompletionTracking": false,
  "taskCompletionUseEmojiShorthand": false,
  "taskCompletionText": "completion",
  "taskCompletionDateFormat": "yyyy-MM-dd",
  "warnOnEmptyResult": true,
  "refreshEnabled": true,
  "refreshInterval": 2500,
  "defaultDateFormat": "MMMM dd, yyyy",
  "defaultDateTimeFormat": "h:mm a - MMMM dd, yyyy",
  "maxRecursiveRenderDepth": 4,
  "tableIdColumnName": "File",
  "tableGroupColumnName": "Group",
  "allowHtml": true,
  "inlineQueryPrefix": "=",
  "inlineJsQueryPrefix": "$=",
  "inlineQueriesInCodeblocks": true,
  "enableDataviewJs": true,
  "enableInlineDataviewJs": false,
  "prettyRenderInlineFields": true
}
```

**Comment by kearalong** [comment]
Mine also hasn't been fixed in the update. The only table views that work for me are still ones mapped to a single tag, not classified in front matter or if they have more than one tag.

**Comment by mdelobelle** [comment]
Thanks to @Krakor92 we have pushed a new version (0.3.9). Does it fix your issue?

**Comment by balola** [comment]
Hi,

My problem is, I want to save date information in YYYY.MM format:

- I tried in input field, but 2022.10 changed to 2022.1
tól:: {"type":"Input","options":{}}

- Then in date field, the setting was
![image](https://user-images.githubusercontent.com/24825268/203257723-f2cd6044-d30e-4915-a37b-d195fe888064.png)

but the date data disappeared from the form
![image](https://user-images.githubusercontent.com/24825268/203257552-e4c21da4-cd13-4f27-bd3a-22982f3c2260.png)

- In the table view I saw in normal date format (OK, it is not a real problem)
![image](https://user-images.githubusercontent.com/24825268/203258690-18469004-7740-43a0-882c-9adb6e2b1bae.png)

- but it is hard to write old yearly data (1967.01.01) and the month combo box doesn't open down.
- the File is not a link, just seems to 
- and a special problem, I used to save Outlook email data in input field **as link**, so I drag-and-drop emails into task input field, which is in an ".msg" file extension (clicking on it I can open the Email). But I cannot do it into your modal form
![image](https://user-images.githubusercontent.com/24825268/203262590-53296c6d-e499-4b32-95c9-7b4b2aabdc92.png)


I love your program, it is a great work and improve very fast, but sometimes little things cause big problems.
So, if you have time to think it over, please find solutions for them.

Fülike

**Comment by mdelobelle** [comment]
Hi could you open separate issues for that? i'll close this one that has been fixed by 0.3.9
